### DISCUSSION TITLE
Is there a way to identify the board model/variant in the sketch?

### DISCUSSION THREAD

**Discussion by CMB27** [earlephilhower/arduino-pico#1645] [Q&A] [answered]
I'm looking for something like `#ifdef ARDUINO_SAMD_NANO_33_IOT`, but of course for boards with the RP2040.

**Comment by earlephilhower** [comment, marked as answer]
`ARDUINO_<boardname>` is defined on the `gcc` command line.  Check `boards.txt` for the exact `<variant>.build.board` line (but it's normally pretty obvious).

**Reply by CMB27** [reply]
That worked.
Thank you
